gateway-server/src/main/java/org/apache/knox/gateway/websockets/JWTValidator.java

```diff
@@ -0,0 +1,288 @@
+        TopologyService ts = gatewayServices.getService(ServiceType.TOPOLOGY_SERVICE);
+        for (Topology topology : ts.getTopologies()) {
+            if (topology.getName().equals("knoxsso")) {
+                for (Service service : topology.getServices()) {
+                    if (service.getRole().equals("KNOXSSO")) {
+                        params = service.getParams();
+                    }
+                    break;
+                }
+                break;
+            }
+        }
+        return params;
+    }
+    private void configureParameters() {
+        Map<String,String> params = getParams();
+        // token verification pem
+        String verificationPEM = params.get(SSO_VERIFICATION_PEM);
+        // setup the public key of the token issuer for verification
+        if (verificationPEM != null) {
+            try {
+                publicKey = CertificateUtils.parseRSAPublicKey(verificationPEM);
+            } catch (ServletException e){
+                throw new RuntimeException("Failed to obtain public key: "+e);
+            }
+        }
+
+        //provider-level configuration
+        providerParamValue = params.get(TokenStateService.CONFIG_SERVER_MANAGED);
+
+        cookieName = params.get(KNOXSSO_COOKIE_NAME);
+        if (cookieName == null) {
+            cookieName = DEFAULT_SSO_COOKIE_NAME;
+        }
+        expectedIssuer =  params.get(JWT_EXPECTED_ISSUER);
+        if (expectedIssuer == null) {
+            expectedIssuer = JWT_DEFAULT_ISSUER;
+        }
+        expectedSigAlg =  params.get(JWT_EXPECTED_SIGALG);
+        if (expectedSigAlg == null) {
+            expectedSigAlg = JWT_DEFAULT_SIGALG;
+        }
+        authorityService = gatewayServices.getService(ServiceType.TOKEN_SERVICE);
+        if (isServerManagedTokenStateEnabled()) {
+            tokenStateService = gatewayServices.getService(ServiceType.TOKEN_STATE_SERVICE);
+        }
+        // Setup the verified tokens cache
+        signatureVerificationCache = SignatureVerificationCache.getInstance(
+                "knoxsso", new WebSocketFilterConfig(params));
+    }
+
+    private void extractToken(ServletUpgradeRequest req){
+        List<HttpCookie> ssoCookies = req.getCookies();
+        if (ssoCookies != null){
+            for (HttpCookie ssoCookie : ssoCookies) {
+                if (cookieName.equals(ssoCookie.getName())) {
+                    try {
+                        token = new JWTToken(ssoCookie.getValue());
+                        displayableTokenId = Tokens.getTokenIDDisplayText(TokenUtils.getTokenId(token));
+                        displayableToken = Tokens.getTokenDisplayText(token.toString());
+                        websocketLog.debugLog("found token:"+displayableToken+" id:"+displayableTokenId);
+                        return;
+                    } catch (ParseException e) {
+                        // Fall through to keep checking if there are more cookies
```

Comment:
Need a log statement here to indicate what failed.

Reply:
this is not a real failure, just an indication that this cookie doesn't contain the jwt we are looking for. so we continue checking the next cookie